shiny/workflowPlot/server.R

```diff
@@ -98,26 +100,33 @@ server <- shinyServer(function(input, output, session) {
   # Allows to load actual data (different from model output) following the tutorial
   # https://github.com/PecanProject/pecan/blob/develop/documentation/tutorials/AnalyzeOutput/modelVSdata.Rmd
   # @params: bety,settings,File_path,File_format
-  loadObservationData <- function(bety,settings,File_path,File_format){
-    start.year<-as.numeric(lubridate::year(settings$run$start.date))
-    end.year<-as.numeric(lubridate::year(settings$run$end.date))
-    site.id<-settings$run$site$id
+  # loadObservationData <- function(bety,settings,File_path,File_format){
+  loadObservationData <- function(bety,inputs_df){
+    input_id <- inputs_df$input_id
+    File_format <- getFileFormat(bety,input_id)
+    start.year <- as.numeric(lubridate::year(inputs_df$start_date))
+    end.year <- as.numeric(lubridate::year(inputs_df$end_date))
+    File_path <- inputs_df$filePath
+    # TODO There is an issue with the db where file names are not saved properly. 
+    # To make it work with the VM, uncomment the line below
+    # File_path <- paste0(inputs_df$filePath,'.csv')
+    site.id <- inputs_df$site_id
     site<-PEcAn.DB::query.site(site.id,bety$con)
     observations<-PEcAn.benchmark::load_data(data.path = File_path, format= File_format, time.row = File_format$time.row,  site = site, start_year = start.year, end_year = end.year) 
     return(observations)
   }
-  # This function as a wrapper over PEcAn.DB::query.format.vars where 
+  # This function is a wrapper over PEcAn.DB::query.format.vars where 
   # file format can be retrieved using either by input or format id. 
   getFileFormat <- function(bety,input.id,format.id=NULL){
-    # Retaining the code for getting file format using format Id as in tutorial
+    # TODO Retaining the code for getting file format using format Id as in tutorial
```

Comment:
This function is not necessary, since it just directly calls `query.format.vars` with the same arguments. You should remove it here and replace it elsewhere with the direct call.